src/java/ognl/OgnlRuntime.java

```diff
@@ -1966,7 +1971,11 @@ public static Map getFields(Class targetClass)
                     Field fa[];
 
                     result = new HashMap(23);
-                    fa = targetClass.getDeclaredFields();
+                    try {
```

Comment:
Thanks! I got your idea but they aren't in same scope (method). Their call order isn't guaranteed. Furthermore, I'm not going to use a global flag because user might remove security manager at runtime after some calls. At bottom, according to my googling and experience, java security checks aren't expensive.

Reply:
Hi @yasserzamani . I know this PR has already been merged, but I wanted to respond to make sure I understood your reply.

Ooops ... I missed that the two code blocks were in separate methods, and interpreted them as being in the same block scope with a loop involved.  :disappointed_relieved:  Since there's not actually any looping over a try..catch involved the new logic shouldn't introduce any noticeable overhead (especially since post-computation results are cached).

Thanks for clarifying and pointing out my mis-interpretation. :smile:

Reply:
Cool! Frankly I had overlooked the performance. Thanks for your acknowledgement.